src/webClient.ts

```diff
@@ -78,6 +76,7 @@ export function sleepFor(sleepDurationInSeconds: number): Promise<any> {
 
 async function sendRequestInternal(request: WebRequest): Promise<WebResponse | undefined> {
     console.log(util.format("[%s]%s", request.method, request.uri));
+    const httpCallbackClient: httpClient.HttpClient = new httpClient.HttpClient(`${process.env.AZURE_HTTP_USER_AGENT}`, undefined, requestOptions);

```

Comment:
This will create new instance of httpclient every time a request is made. Instead it should be a Singleton class.